README.md

```diff
@@ -49,29 +49,136 @@ import { default as strava, Strava } from 'strava-v3';
+**After Migration:**
+
+```javascript
+const strava = require('strava-v3');
+
+// Configuring Strava with necessary credentials
+strava.config({
+  access_token: 'your_access_token',
+  client_id: 'your_client_id',
+  client_secret: 'your_client_secret',
+  redirect_uri: 'your_redirect_uri'
+});
+
+// Fetching athlete data using Promises with async/await
+async function fetchAthleteData(athleteId) {
+  try {
+    const athlete = await strava.athlete.get({ id: athleteId });
+    console.log('Athlete Data:', athlete);
+    
+    // Fetching athlete's activities using Promises with async/await
+    const activities = await strava.activities.list({ athlete_id: athlete.id });
+    console.log('Activities:', activities);
+    
+  } catch (err) {
+    console.error('Error fetching data:', err);
+  }
+}
+
+// Invoke the async function
+fetchAthleteData(12345);
+```
+
+### Key Changes Explained
+1. **Removal of Callback Parameters:**
+
+    * Before: Methods like `strava.athlete.get` and `strava.activities.list` accept a callback function as the last parameter.
+    * After: These methods now return Promises, eliminating the need for callback functions.
+
+2. **Using `async/await`:**
+
+    * The `fetchAthleteData` function is declared as `async`, allowing the use of `await` to handle Promises in a synchronous-like manner.
+    * `await` pauses the execution of the function until the Promise resolves, making the code easier to read and maintain.
+
+3. **Centralized Error Handling:**
+
+    * The `try-catch` block encapsulates both asynchronous operations, ensuring that any errors thrown by either `strava.athlete.get` or `strava.activities.list` are caught and handled in one place.
+
+4. **Elimination of Nested Callbacks:**
+
+    * The Promise-based approach avoids deeply nested functions, reducing complexity and improving readability.
+
+### Alternative: Using `.then()` and `.catch()`
+If you prefer not to use `async/await`, you can achieve similar results using `.then()` and `.catch()` chaining.
+
+**Example:**
+
+```javascript
+const strava = require('strava-v3');
+
+// Configuring Strava with necessary credentials
+strava.config({
+  access_token: 'your_access_token',
+  client_id: 'your_client_id',
+  client_secret: 'your_client_secret',
+  redirect_uri: 'your_redirect_uri'
+});
+
+// Fetching athlete data using Promises with .then() and .catch()
+strava.athlete.get({ id: 12345 })
+  .then(athlete => {
+    console.log('Athlete Data:', athlete);
+    return strava.activities.list({ athlete_id: athlete.id });
+  })
+  .then(activities => {
+    console.log('Activities:', activities);
+  })
+  .catch(err => {
+    console.error('Error fetching data:', err);
+  });
+  ```
+**Benefits:**
+
+* **Chainable:** Promises allow you to chain multiple asynchronous operations in a linear and manageable fashion.
+* **Error Propagation:** Errors thrown in any `.then()` block are propagated down the chain to the nearest `.catch()` block.
+
```

Comment:
You are right that `.then()` and `.catch()` can be used instead, but I would not recommend them. 

My understanding of the history is that these were introduced first, but once `async/await` were added to the language, it's more natural to use `await` instead of `.then()`, and using `throw` unifies async and sync error handling, which is great.